src/FormControl/FormControl.tsx

```diff
@@ -232,4 +152,5 @@ export default Object.assign(FormControl, {
   Label: FormControlLabel,
   LeadingVisual: FormControlLeadingVisual,
   Validation: FormControlValidation,
+  useForwardedProps: useFormControlForwardedProps,
```

Comment:
It's debatable whether this should even be publicly exposed with this approach:


On one hand, this is almost never necessary. Prior to this change, wrapping a Primer form input in another component (ie, if you want to make a `NumericTextInput` component) would have broken the `FormControl` wiring. But with context, this doesn't happen. As long as the underlying component is a Primer form input, it doesn't matter how deeply nested it is. If we don't expose this hook, we can push users to always build around Primer components and not HTML from scratch.

On the other hand, publicly exposing it allows consumers to build custom form inputs from plain HTML. Maybe that's not something we want to prevent. For example, they could build a new `RangeInput` component, which Primer doesn't have yet.

Reply:
I feel more strongly about exposing this after looking into the ongoing work around `Filter` and `QueryBuilder` - more complex inputs that definitely could benefit from being able to directly access this info.

Reply:
I'm definitely a fan of making this public, depending on how stable this ends up being maybe it makes sense in the drafts/experimental entrypoint until it's further validated?

I think something that was surprising initially was seeing this on the component namespace. Could hooks be something that is just accessed from the entrypoint instead of tying it to a specific component? e.g. `import { useFormControl } from '@primer/react';` or is the namespace helpful here in the same way that it helps with sub-components?

Reply:
We definitely could expose the hook directly - I just figured the namespace was an obvious place to look for it. I am good with either one

Reply:
I slightly prefer exposing the hook directly. Usually the properties on a component "object" are other components.